dds/DCPS/XTypes/TypeAssignability.h

```diff
@@ -118,8 +89,12 @@ class OpenDDS_Dcps_Export TypeAssignability {
   bool get_struct_member(const MinimalTypeObject*& ret, const CommonStructMember& m) const;
   bool get_union_member(const MinimalTypeObject*& ret, const CommonStructMember& m) const;
 
-private:
-  TypeLookup typelookup_;
+  TypeLookupService* tl_service_;
```

Comment:
Do you mean to ask if `TypeLookupService` is designed to be copied?

Reply:
`TypeAssignability` doesn't have any internal state other than `TypeLookupService`, so I think whether copying is allowed depends on how we're going to use `TypeLookupService`.

Reply:
> Do you mean to ask if `TypeLookupService` is designed to be copied?

I didn't.

Unless I missed something, the TypeAssignability class is copyable but contains a pointer to TypeLookupService.  This is a problem because it assumes the lifetime of TypeLookupService exceeds that of (all copies of) TypeAssignability.  Using a smart pointer would help this.  Or if TypeAssignability's use was more constrained (as in, couldn't be copied and we can make guarantees about its lifetime relative to TypeLookupService), this could be a reference or at least a const pointer.

Reply:
I've changed it to the smart pointer.